File: RealTimePortal.UI/src/app/processes/models/process.ts
export interface Process {
  id: number;
  executionId: string;

  clientId: string;
  applicationId: string;
  processType: string;
  referenceNumber: string;

  status: string | number;

  startedByUserId: number | null;
  startedAt: string | null;
  completedAt: string | null;

  errorMessage: string | null;
  requestData: string | null;

  createdAt: string;
  createdBy: string | null;
  modifiedAt: string | null;
  modifiedBy: string | null;

  steps: ProcessStep[];
}

export interface ProcessStep {
  id: number;
  processId: number;
  stepNumber: number;
  stepName: string;

  status: string | number;
  progressPercentage: number;
  retryCount: number;

  message: string | null;
  inputData: string | null;
  outputData: string | null;

  startedAt: string | null;
  completedAt: string | null;
  errorMessage: string | null;
}

File: RealTimePortal.Domain/Process.cs
namespace RealTimePortal.Domain;

public class Process
{
    public long Id { get; set; }
    public string ExecutionId { get; set; } = string.Empty;
    public string ClientId { get; set; } = string.Empty;

    public string ApplicationId { get; set; } = string.Empty;

    public string ProcessType { get; set; } = string.Empty;

    public string ReferenceNumber { get; set; } = string.Empty;

    public ProcessStatus Status { get; set; }

    public long? StartedByUserId { get; set; }

    public DateTime? StartedAt { get; set; }

    public DateTime? CompletedAt { get; set; }

    public string? ErrorMessage { get; set; }

    public string? RequestData { get; set; }

    public DateTime CreatedAt { get; set; }

    public string? CreatedBy { get; set; }

    public DateTime? ModifiedAt { get; set; }

    public string? ModifiedBy { get; set; }

    public ICollection<ProcessStep> Steps { get; set; }
        = new List<ProcessStep>();
}